### PR TITLE
Add input types to attributes

### LINKED ISSUE #4254: Dashboard 2.0: Ability to create attributes with different data types for attribute values
Depends on: #4244 

Add ability to create attributes with different data types for attribute values. Requires the following changes in Dashboard 2.0:

On the attribute details page:
- [x] Add input to allow selecting the input type for an attribute.
- [ ] Adjust adding values of an attribute to depend on the selected input type.

On the product details page:
- [ ] Support rendering different inputs for attributes, based on their configuration.

### PR DESCRIPTION
**Pull request by dominik-zeglen**
I want to merge this change because it adds input types to new attributes.
Resolves #4254

### Pull Request Checklist

1. [ ] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Database queries are optimized and the number of queries is constant.
1. [ ] Database migration files are up to date.
1. [ ] The changes are tested.
1. [ ] The code is documented (docstrings, project documentation).
1. [ ] GraphQL schema and type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### CODE REVIEW

**Comment by django-queries** [comment]
Here is the report for 30279ee836c5b619b25709663861200512ac116e (mirumee/saleor @ attributes/add-input-types)
Base comparison is 7cdebe3efa8f95373791756856917cfb6063613a.



<details><summary>No differences were found. (click me)</summary>
<p>

```diff
# api.benchmark checkout
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  add billing address to checkout     	         41	         41	             24
  add shipping to checkout            	          7	          7	              0
  checkout payment charge             	         16	         16	              0
  complete checkout                   	          6	          6	              0
  create checkout                     	         45	         45	             21

# api.benchmark homepage
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  retrieve main menu                  	          5	          5	              0
  retrieve product list               	          4	          4	              0
  retrieve secondary menu             	          5	          5	              0
  retrieve shop                       	          2	          2	              0

# api.benchmark product
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  product details                     	         15	         15	              3

# api.benchmark variant
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  retrieve variant list               	         15	         15	              6
```

</p>
</details>

File: saleor/static/dashboard-next/storybook/stories/products/ProductUpdatePage.tsx [hunk 1]
```diff
@@ -1,117 +1,66 @@
...
+  images: product.images,
+  onAttributesEdit: () => undefined,
+  onBack: () => undefined,
+  onDelete: () => undefined,
+  onImageDelete: () => undefined,
+  onImageUpload: () => undefined,
+  onProductShow: () => undefined,
+  onSubmit: () => undefined,
+  onVariantAdd: () => undefined,
+  onVariantShow: () => undefined,
```
Comment (piotrgrundas) [inline saleor/static/dashboard-next/storybook/stories/products/ProductUpdatePage.tsx]: Why not void?

**Comment by NyanKiyoshi** [comment]
When we create an attribute value, it looks like the admin value is the `value` field (which is a meta, e.g. a color code) and when the dashboard renders the returned admin value by GQL, it shows the slug.

Didn't you mean to use the `slug` field instead of the `value` field?

**Comment by dominik-zeglen** [comment]
Not a part of this PR but noted

**Pull request merged**